DownFall/Views/MenuSpriteNode.swift

```diff
@@ -35,44 +39,184 @@ class MenuSpriteNode: SKSpriteNode {
+        removeAllChildren()
+        
+        let border = SKShapeNode(rect: self.frame)
+        border.strokeColor = UIColor.darkGray
+        border.lineWidth = 10.0
+        addChild(border)
+        
         zPosition = precedence.rawValue
+        setupButtons(menuType, playableRect, precedence: precedence)
+        
+    }
+    
+    private func setupButtons(_ menuType: MenuType, _ playableRect: CGRect, precedence: Precedence) {
+        let menuSizeWidth = playableRect.size.width * menuType.widthCoefficient
+        let menuSizeHeight = playableRect.size.height * menuType.heightCoefficient
+        let buttonSize = CGSize(width: menuSizeWidth * 0.4, height: 120)
+        
+        var addDefaultButton = true
+        
+        if menuType == .rotate {
+            
+            // Heights and widths
+            let topHeight: CGFloat = menuSizeHeight/3 - 56
+            let bottomHeight: CGFloat = -menuSizeHeight/3 + 56
+            
+            let rightWidth: CGFloat = menuSizeWidth/2 - 56
+            let leftWidth: CGFloat = -menuSizeWidth/2 + 56
+            
+            // Points
+            let topRight = CGPoint(x: rightWidth, y: topHeight)
+            let topLeft = CGPoint(x: leftWidth, y: topHeight)
+            
+            let bottomRight = CGPoint(x: rightWidth, y: bottomHeight)
+            let bottomLeft = CGPoint(x: leftWidth, y: bottomHeight)
+            
+            let startStopTarget: [(CGPoint, CGPoint)] = [
+                (topRight, topLeft),
+                (topLeft, bottomLeft),
+                (bottomLeft, bottomRight),
+                (bottomRight, topRight)
+            ]
+            
+            var actions: [SKAction] = []
+            
+            let fingerSprite = SKSpriteNode(texture: SKTexture(imageNamed: "finger"))
+            fingerSprite.size = CGSize(width: 80, height: 80)
+            
+            // initial positions
+            fingerSprite.position = topRight
+            fingerSprite.zPosition = precedence.rawValue
+            
+            // add to scene
+            addChild(fingerSprite)
+            
+            for startStop in startStopTarget {
+                let stop = startStop.1
+                
+                //actions
+                let swipeAction = SKAction.move(to: stop, duration: 1.25)
+                swipeAction.timingMode = .easeOut
+                
+                actions.append(swipeAction)
+            }
+            
+            fingerSprite.run(SKAction.repeatForever(SKAction.sequence(actions)))
+            
+            let paragraphNode = ParagraphNode.labelNode(text: "Swipe like this to rotate the board counter-clockwise", paragraphWidth: playableRect.width * 0.70)
+            paragraphNode.position = .zero
+            paragraphNode.zPosition = precedence.rawValue
+            
+            addChild(paragraphNode)
+            
+        } else if menuType == .gameWin && GameScope.shared.difficulty == .tutorial1 {
```

Comment:
Should this just be a separate menu type rather than depending on two distinct variables?